proof/crefine/RISCV64/Recycle_C.thy

```diff
@@ -908,16 +1027,20 @@ sorry (* FIXME RT: cancelBadgedSends_ccorres *) (*
                                     cmachine_state_relation_def)
               apply (clarsimp simp: cpspace_relation_def typ_heap_simps'
                                     update_ep_map_tos)
-              apply (erule(1) cpspace_relation_ep_update_ep2)
-               apply (simp add: cendpoint_relation_def Let_def)
-               apply (subgoal_tac "tcb_at' (last (a # list)) \<sigma> \<and> tcb_at' a \<sigma>")
-                apply (clarsimp simp: is_aligned_neg_mask_eq[OF is_aligned_tcb_ptr_to_ctcb_ptr[where P=\<top>]])
-                apply (simp add: tcb_queue_relation'_def EPState_Send_def mask_def)
-                apply (drule (1) tcb_and_not_mask_canonical[where n=2])
-                 apply (simp (no_asm) add: tcbBlockSizeBits_def)
-                subgoal by (simp add: mask_def canonical_bit_def)
-               subgoal by (auto split: if_split)
-              subgoal by simp
+              apply (rule conjI)
+               apply (erule(1) cpspace_relation_ep_update_ep2)
+                apply (simp add: cendpoint_relation_def Let_def)
+                apply (subgoal_tac "tcb_at' (last (a # list)) \<sigma> \<and> tcb_at' a \<sigma>")
```

Comment:
reading this proof, I have no idea what list, a, lista, and xs are ... any chance you could rename things a bit to make it easier to follow, or add some commentary?

Reply:
I'm pretty sure that these are related to the endpoint queue, but this is a part of the proof that I didn't change; I did try to keep as much of the old proof as possible and didn't really look into improving it. But sure, I can investigate and see whether a nicer renaming can be done

Reply:
If you didn't change it, then might as well leave it. For some reason I thought it was new.

Reply:
Alrighty. Often I do need to reindent parts of the old proof, and I think that's what happened here.